Tests/Common/Orders/OrderSizingTests.cs

```diff
@@ -0,0 +1,141 @@
+﻿/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using System;
+using NUnit.Framework;
+using QuantConnect.Algorithm.Framework.Portfolio;
+using QuantConnect.Data.Market;
+using QuantConnect.Orders;
+using QuantConnect.Orders.Fees;
+using QuantConnect.Tests.Common.Securities;
+using QuantConnect.Tests.Engine.DataFeeds;
+
+namespace QuantConnect.Tests.Common.Orders
+{
+    [TestFixture]
+    public class OrderSizingTests
+    {
+        [TestCase(0.98, 0)]
+        [TestCase(-0.98, 0)]
+        [TestCase(0.9999999, 1)]
+        [TestCase(-0.9999999, -1)]
+        public void AdjustByLotSize(decimal quantity, decimal expected)
+        {
+            var algo = new AlgorithmStub();
+            var security = algo.AddEquity(Symbols.SPY.Value);
+
+            var result = OrderSizing.AdjustByLotSize(security, quantity);
+
+            Assert.AreEqual(expected, result);
+        }
+
+        [Test]
+        public void GetOrderSizeForPercentVolume()
+        {
+            var algo = new AlgorithmStub();
+            var security = algo.AddFutureContract(Symbols.Future_CLF19_Jan2019);
+            security.SetMarketPrice(new TradeBar { Value = 250, Volume = 10});
+
+            var result = OrderSizing.GetOrderSizeForPercentVolume(security, 0.5m, 100);
+            Assert.AreEqual(5, result);
+        }
+
+        [TestCase(100000, 100)]
+        [TestCase(1000000, 100)]
+        [TestCase(1000000, 1)]
+        [TestCase(1000000, -1)]
+        public void GetOrderSizeForMaximumValue(decimal maximumOrderValue, decimal target)
```

Comment:
Please include expected results in test case

Reply:
Good idea, addressed 👍